src/orchestrator-kubernetes/src/cloud_resource_controller.rs

```diff
@@ -103,6 +106,63 @@ impl CloudResourceController for KubernetesOrchestrator {
         Ok(endpoints)
     }
 
+    async fn watch_vpc_endpoints(&self) -> BoxStream<'static, VpcEndpointEvent> {
+        let stream = watcher(self.vpc_endpoint_api.clone(), watcher::Config::default())
+            .touched_objects()
+            .filter_map(|object| async move {
+                match object {
+                    Ok(vpce) => {
+                        if let Some(connection_id) =
+                            mz_cloud_resources::id_from_vpc_endpoint_name(&vpce.name_any())
+                        {
+                            if let Some(state) =
+                                vpce.status.as_ref().and_then(|st| st.state.to_owned())
+                            {
+                                return Some(VpcEndpointEvent {
+                                    connection_id,
+                                    status: state,
+                                    // Use the 'Available' Condition on the VPCE Status to set the event-time, falling back
+                                    // to now if it's not set
+                                    time: vpce
+                                        .status
+                                        .unwrap()
+                                        .conditions
+                                        .and_then(|c| {
+                                            c.into_iter().find(|c| &c.type_ == "Available")
+                                        })
+                                        .and_then(|condition| {
+                                            Some(condition.last_transition_time.0)
+                                        })
+                                        .unwrap_or_else(Utc::now),
+                                });
+                            } else {
+                                // The Status/State is not yet populated on the VpcEndpoint, which means it was just
+                                // initialized and hasn't yet been reconciled by the environment-controller
+                                // We return an event with an 'unknown' state so that watchers know the VpcEndpoint was created
+                                // even if we don't yet have an accurate status
+                                return Some(VpcEndpointEvent {
+                                    connection_id,
+                                    status: VpcEndpointState::Unknown,
+                                    time: vpce.creation_timestamp().expect("Will exist").0,
```

Comment:
is it documented that it will exist? typically I only use `expect`'s when there is a _local_, control-flow reason it will exist; just in case, you can just use `?` here

Reply:
Will update to use `?` -- according to the docs it is guaranteed to exist since this object was received from the kube api server: https://docs.rs/kube/latest/kube/trait.ResourceExt.html#tymethod.creation_timestamp